traversal/predicate/Predicate.java

```diff
@@ -95,8 +95,12 @@ public boolean apply(AttributeVertex<?> vertex, Traversal.Parameters.Value value
             return argument.apply(operator, vertex, value);
         }
 
-        public boolean apply(ARG_TYPE lhs, ARG_TYPE rhs) {
-            return argument.apply(operator, lhs, rhs);
+        public <T> boolean apply(AttributeVertex<T> vertex, ARG_TYPE rhs) {
+            return apply(vertex.valueType(), vertex.value(), rhs);
+        }
+
+        public <T> boolean apply(Encoding.ValueType<T> lhsType, T lhs, ARG_TYPE rhs) {
```

Comment:
new API on `Predicate` lets us decouple the `Preidcate` from having to be used from an `AttributeVertex` or a `Traversal.Params`, and enables the completely generic execution of a language predicate on two values